common/protos/proto/share.proto

```diff
@@ -0,0 +1,60 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package databend_proto;
+
+import "metadata.proto";
+
+message ShareNameIdent {
+  uint64 ver = 100;
+  uint64 min_compatible = 101;
+
+  // The user this share belongs to
+  string tenant = 1;
+
+  // share name
+  string share_name = 2;
+}
+
+message ShareGrantObject {
+  uint64 ver = 100;
+  uint64 min_compatible = 101;
+
+  oneof object {
+    DatabaseNameIdent database = 1;
+    TableNameIdent table = 2;
+  }
```

Comment:
Isn't there a consistency issue if a `share` identifies a DB or table by name but not by id?